src/resolve/resolved-dns-server.c

```diff
@@ -878,6 +878,13 @@ DnsServer *manager_get_dns_server(Manager *m) {
         if (!m->current_dns_server)
                 manager_set_dns_server(m, m->dns_servers);
 
+        while (m->current_dns_server &&
+               manager_server_is_stub(m, m->current_dns_server)) {
+                manager_next_dns_server(m, NULL);
+                if (m->current_dns_server == m->dns_servers)
```

Comment:
I'm not sure if this is good enough btw. I mean, is it possible that when the loop began, `current_dns_server` is not `dns_servers` (the beginning of the list)?

I thought of remembering the `current_dns_server` before loop begins, but it seems like a server could be "unlinked from the list"? In which case the loop will become infinite?

Reply:
Maybe I should add `|| manager_server_is_stub(m, m->current_dns_server)` here?